File: BACK/API.Library/Infrastructure/Library.Infrastructure/Entities/Items/Movie.cs
using Library.Infrastructure.Entities.Enumerations;

namespace Library.Infrastructure.Entities.Items;

public class Movie : Item
{
    public Movie(string label, DateTime releaseDate, string? support, string? imageURL, string director,
        string editor) : base(label, releaseDate, support, imageURL)
    {
        Type = ItemType.Movie;
        Director = director;
        Editor = editor;
    }

    public string Director { get; set; }
    public string Editor { get; set; }
}

File: FRONT/APP.Mobile/vaultapp/lib/app/modules/items/data/models/movie.dart
import 'package:vaultapp/app/modules/items/data/models/item.dart';

class Movie extends Item {
  Movie(
    {super.id,
    required super.label,
    required super.type,
    required super.releaseDate, 
    super.support,
    super.imageURL,
    required this.director,
    required this.editor});

  final String director;
  final String editor;

  factory Movie.fromJson(Map<String, dynamic> json) {
    String? id = json['id'];
    String label = json['label'];
    String type = json['type'];
    String? support = json['support'];
    String? imageURL = json['imageURL'];
    String director = json['director'];
    String editor = json['editor'];

    DateTime releaseDate = DateTime.parse(json['releaseDate']);
    return Movie(
        id: id,
        label: label,
        type: type,
        releaseDate: releaseDate,
        support: support,
        imageURL: imageURL,
        director: director,
        editor: editor);
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'label': label,
      'type': type,
      'releaseDate': releaseDate.toIso8601String(),
      'support': support,
      'imageURL': imageURL,
      'director': director,
      'editor': editor
    };
  }
}